crates/runner/src/lib.rs

```diff
@@ -10,10 +10,93 @@ use wasi_common::sync::WasiCtxBuilder;
 use wasi_common::WasiCtx;
 use wasmtime::{Config, Engine, Linker, Module, OptLevel, Store};
 
+pub struct Builder {
+    /// The JS source.
+    input: PathBuf,
+    /// Root path. Used resolve the absolute path of the JS source.
+    root: PathBuf,
+    /// `javy` binary path.
+    bin_path: String,
+    /// The path to the wit file.
+    wit: Option<PathBuf>,
+    /// The name of the wit world.
+    world: Option<String>,
+    /// The logger capacity, in bytes.
+    capacity: usize,
+    built: bool,
+}
+
+impl Default for Builder {
+    fn default() -> Self {
+        Self {
+            capacity: usize::MAX,
+            input: PathBuf::from("identity.js"),
+            wit: None,
+            world: None,
+            bin_path: "javy".into(),
+            root: Default::default(),
+            built: false,
+        }
+    }
+}
+
+impl Builder {
+    pub fn root(&mut self, root: impl Into<PathBuf>) -> &mut Self {
+        self.root = root.into();
+        self
+    }
+
+    pub fn input(&mut self, path: impl Into<PathBuf>) -> &mut Self {
+        self.input = path.into();
+        self
+    }
+
+    pub fn bin(&mut self, bin: impl Into<String>) -> &mut Self {
+        self.bin_path = bin.into();
+        self
+    }
+
+    pub fn wit(&mut self, wit: impl Into<PathBuf>) -> &mut Self {
```

Comment:
Arguably wit and world could be one function given they either both need to be provided or neither should be provided but this is fine